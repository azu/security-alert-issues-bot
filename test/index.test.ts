// You can import your modules
// import index from '../src/index'

import nock from "nock";
// Requiring our app implementation
import myProbotApp from "../src";
import {Probot} from "probot";
// Requiring our fixtures
import repository_vulnerability_alert_payload from "./fixtures/repository_vulnerability_alert.create.json";

nock.disableNetConnect();

describe("security-alert-issues-bot", () => {
    let probot: any;

    beforeEach(() => {
        probot = new Probot({id: 123, cert: "test"});
        // Load our app into probot
        const app = probot.load(myProbotApp);

        // just return a test token
        app.app = () => "test";
    });

    test("creates new issue when receive repository_vulnerability_alert.create", async done => {
        // Test that we correctly return a test token
        nock("https://api.github.com")
            .post("/app/installations/2/access_tokens")
            .reply(200, {token: "test"});
        nock("https://api.github.com")
            .get("/repos/azu/.github/contents/.github/security-alert-issues-bot.yml")
            .reply(404, {
                "message": "Not Found",
                "documentation_url": "https://developer.github.com/v3/repos/contents/#get-contents"
            });
        nock("https://api.github.com")
            .get("/repos/azu/github-webhook-SecurityVulnerability-test/contents/.github/security-alert-issues-bot.yml")
            .reply(404, {
                "message": "Not Found",
                "documentation_url": "https://developer.github.com/v3/repos/contents/#get-contents"
            });
        // Test that a comment is posted
        nock("https://api.github.com")
            .post("/repos/azu/github-webhook-SecurityVulnerability-test/issues", (body: any) => {
                done(expect(body).toMatchSnapshot());
                return true;
            })
            .reply(200);

        // Receive a webhook event
        await probot.receive({name: "repository_vulnerability_alert", payload: repository_vulnerability_alert_payload});
    });
});
