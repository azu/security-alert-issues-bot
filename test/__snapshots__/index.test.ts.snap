// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`security-alert-issues-bot creates a comment when an issue is opened 1`] = `
Object {
  "body": "
- Affected Package: [lodash](https://www.npmjs.com/package/lodash) < 4.17.5
- Fixed in [4.17.5](https://npmfs.com/package/lodash/4.17.5/)

**How to fix?**

Upgrade lodash to version 4.17.5 or later. For example:

\`\`\`
\\"dependencies\\": {
  \\"lodash\\": \\">=4.17.5\\"
}
\`\`\`

or…

\`\`\`
\\"devDependencies\\": {
  \\"lodash\\": \\">=4.17.5\\"
}
\`\`\`

## Details

- [CVE-2018-3721](https://nvd.nist.gov/vuln/detail/CVE-2018-3721)

",
  "labels": Array [
    "security",
  ],
  "title": "Vulnerability found in lodash < 4.17.5",
}
`;

exports[`security-alert-issues-bot creates new issue when receive repository_vulnerability_alert.create 1`] = `
Object {
  "body": "
- Affected Package: [lodash](https://www.npmjs.com/package/lodash) < 4.17.5
- Fixed in [4.17.5](https://npmfs.com/package/lodash/4.17.5/)

**How to fix?**

Upgrade lodash to version 4.17.5 or later. For example:

\`\`\`
\\"dependencies\\": {
  \\"lodash\\": \\">=4.17.5\\"
}
\`\`\`

or…

\`\`\`
\\"devDependencies\\": {
  \\"lodash\\": \\">=4.17.5\\"
}
\`\`\`

## Details

- [CVE-2018-3721](https://nvd.nist.gov/vuln/detail/CVE-2018-3721)

",
  "labels": Array [
    "security",
  ],
  "title": "Vulnerability found in lodash < 4.17.5",
}
`;
